src/libs/data-utils.js

```diff
@@ -269,90 +268,130 @@ export const EntityUploader = class EntityUploader extends Component {
+          'where ',
+          span({ style: { fontFamily: 'monospace', fontWeight: '600' } }, ['[type]']),
+          ` is the desired name of the data table in ${getAppName()}. For example, to create or update a `,
+          span({ style: { fontFamily: 'monospace', fontWeight: '600' } }, ['participant']),
+          ' table use ',
+          span({ style: { fontFamily: 'monospace', fontWeight: '600' } }, ['entity:participant_id']),
+          div({ style: { marginTop: '0.5rem', marginBottom: '0.5rem' } }, ['All of the values in the ID column must be unique.']),
+          div({ style: { height: '0.5rem' } }),
+          h(SimpleTabBar, {
+            style: { paddingTop: '0.5rem' },
+            tabs: [{ title: 'File Import', key: true, width: 121 }, { title: 'Text Import', key: false, width: 127 }],
+            value: isFileImportCurrMode,
+            onChange: value => {
+              this.setState({ isFileImportCurrMode: value })
+            }
+          }),
+          div({
+            style: {
+              padding: '1rem 0 0',
+              height: '3.25rem'
+            }
+          }, [
+            isFileImportCurrMode ? div([
+              'Select the ',
+              h(TooltipTrigger, { content: 'Tab Separated Values', side: 'bottom' },
+                [span({ style: { textDecoration: 'underline dashed' } }, 'TSV')]),
+              ' file containing your data: '
+            ]) : div(['Paste the data directly below:']),
+            currentFile && div({ style: { display: 'flex', justifyContent: 'flex-end' } }, [
+              h(Link,
+                {
+                  onClick: () => {
+                    this.setState({ fileContents: '', file: undefined, useFireCloudDataModel: false })
+                  }
+                }, ['Clear'])
+            ])
+          ]),
+          isFileImportCurrMode ? div([
+            h(Clickable, {
+              style: {
+                ...Style.elements.card.container, flex: 1, backgroundColor: dragging ? colors.accent(0.2) : colors.dark(0.1),
+                border: isInvalid ? `1px solid ${colors.danger()}` : `1px dashed ${colors.dark(0.7)}`, boxShadow: 'none'
+              },
+              onClick: openUploader
+            }, [div(['Drag or ', h(Link, ['Click']), ' to select a .tsv file'])]),
+            div({ style: { paddingTop: '0.5rem' } }, [
+              'Selected File: ',
+              span({ style: { color: colors.dark(1), fontWeight: 550 } },
+                (currentFile && currentFile.name) ? currentFile.name : 'None')
+            ])
+          ]) : div([
+            h(PasteOnlyInput, {
+              'aria-label': 'Paste text data here',
+              placeholder: 'entity:participant_id(tab)column1(tab)column2...',
+              onPaste: pastedText => {
+                this.setState({ file: new File([pastedText], 'upload.tsv'), fileContents: pastedText, isFileImportLastUsedMode: false })
+              },
+              readOnly: !!fileContents,
```

Comment:
TOL: React currently prints a warning about the usage pattern of `onChange` and `readOnly` here. I understand the reasoning for doing it this way, but we should revisit this afterward and see if there's another way to accomplish that.